sdk/core/azure-core/azure/core/pipeline/policies/_universal.py

```diff
@@ -104,6 +104,56 @@ def on_request(self, request):
         if additional_headers:
             request.http_request.headers.update(additional_headers)
 
+class RequestIdPolicy(SansIOHTTPPolicy):
+    """A simple policy that sets the given request id in the header.
+
+    This will overwrite request id that is already defined in the request. Request id can be
+    configured up front, where the request id will be applied to all outgoing
+    operations, and additional request id can also be set dynamically per operation.
+
+    :param dict base_headers: Headers to send with the request.
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/test_example_sansio.py
+            :start-after: [START request_id_policy]
+            :end-before: [END request_id_policy]
+            :language: python
+            :dedent: 4
+            :caption: Configuring a request id policy.
+    """
+    def __init__(self, request_id=None, **kwargs):  # pylint: disable=super-init-not-called
+        # type: (str, dict) -> None
+        self._request_id = request_id
+
+    @property
+    def request_id(self):
+        """The current headers collection."""
+        return self._request_id
+
+    def set_request_id(self, value):
+        """Add the request id to the configuration to be applied to all requests.
+
+        :param str value: The request id value.
+        """
+        self._request_id = value
+
+    def on_request(self, request):
+        # type: (PipelineRequest) -> None
+        """Updates with the given request id before sending the request to the next policy.
+
+        :param request: The PipelineRequest object
+        :type request: ~azure.core.pipeline.PipelineRequest
+        """
+        request_id = request.context.options.pop('request_id', None)
```

Comment:
**If** we supported a `request_id` as a constructor parameter, why not use that as the default value rather than having the ifs below?